server/src/routes/routes.js

```diff
@@ -1,10 +1,14 @@
 
 import express from 'express';
+import userValidator from '../validators/userValidator';
+import userFieldsErrors from '../middleware/userFieldsErrors';
+import userController from '../controllers/userController';
 
 const router = express.Router();
 
 router.get('/ping', (req, res) => {
   res.json({ status: 'working' });
 });
+router.post('/api/signup', userValidator, userFieldsErrors, userController.createNewUser);
```

Comment:
Its good call that you are using userValidator and userFieldErrors middleware over here.
But if you are adding restrictions on the db fields then it's better that db checks the object itself and then throws the error.
You can provide all the validations like length, required with the mongoose schema itself, you don't have to check it upfront with another library

Reply:
I actually like this approach better, because it enables us to send user-friendly error messages to client.

I mean, if we let mongodb throw an error, we'll have to parse the error message and send the user-friendly message to client. Or send something generic like "there was an error".

Also, if there are more than one validation errors, user can see all of them at once because we're sending errors array. This is not possible in database level validation I think.

Reply:
I agree with Nitin.